Update Makefile for Multi-Arch build

fdelucchijr opened the pull request:
Using  [vinodkumark369](https://github.com/vinodkumark369) script. Tested in WSL.

vinodkumark369 commented:
@fdelucchijr It is not my script, that is already present in the github action for release of bin image. That is why I am confused about how the image is actually built. I am assuming it is built using circle-ci.

fdelucchijr commented:
> That is already present in the github action for release of bin image.

🤔, I think the CD script is .circleci/config.yml, in the last step triggers the makefile I'd change in this PR.

vinodkumark369 commented:
@fdelucchijr I think this should be enough to build a multi-arch image. My only concern will be the image `okteto/golang-ci:1.16` used in the release job. That image is also build for `linux/amd64` only. I don't know what it is supposed to do, so maybe @pchico83 could check that.

vinodkumark369 commented:
@fdelucchijr @pchico83 Do we really need the Makefile anymore? Could that just be a step inside the release job?

pchico83 commented:
> @fdelucchijr I think this should be enough to build a multi-arch image. My only concern will be the image `okteto/golang-ci:1.16` used in the release job. That image is also build for `linux/amd64` only. I don't know what it is supposed to do, so maybe @pchico83 could check that.

Since that image is just running the `docker build` command, I think it's fine if it's only amd64

pchico83 commented:
> @fdelucchijr @pchico83 Do we really need the Makefile anymore? Could that just be a step inside the release job?

Yeap, +1 for removing `Makefile`

pchico83 commented:
@fdelucchijr you need to update https://github.com/okteto/bin-image/blob/master/.circleci/config.yml#L13 to the tag 1.17.11. We have added the `buildx` plugin to the image

pchico83 commented:
> @fdelucchijr you need to update https://github.com/okteto/bin-image/blob/master/.circleci/config.yml#L13 to the tag 1.17.11. We have added the `buildx` plugin to the image

@fdelucchijr done already here https://github.com/okteto/bin-image/pull/5

fdelucchijr commented:
Hi @pchico83! As I see in Okteto/devenv#96 your trying to build the images in GH actions, I think that’s a wonderful idea and have a lot more experience to contribute. I’ve removed the CircleCI pipeline and added an almost identical GH Actions pipeline we use at our business. 
It pushes the image to the repo registry and to Docker Hub, because of that it needs to get Docker credential from repo secrets. Hope it helps and centralize a little the build process with the repo.

pchico83 commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,60 @@
+
+on:
+  pull_request:
+    branches: [master]
```
we should only build on PRs, and push on merges to `master`

pchico83 commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,60 @@
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
```
this should only be done on merges to `master`, not on PRs

pchico83 commented on .github/workflows/deploy.yaml:
```diff
@@ -0,0 +1,81 @@
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          username: "${{ secrets.DOCKER_USERNAME }}"
```
```suggestion
          username: "${{ secrets.DOCKER_USER }}"
```

pchico83 commented on .github/workflows/deploy.yaml:
```diff
@@ -0,0 +1,81 @@
+        uses: docker/login-action@v1
+        with:
+          username: "${{ secrets.DOCKER_USERNAME }}"
+          password: "${{ secrets.DOCKER_TOKEN }}"
```
```suggestion
          password: "${{ secrets.DOCKER_PASS }}"
```

pchico83 commented on .github/workflows/deploy.yaml:
```diff
@@ -0,0 +1,81 @@
+            ${{ runner.os }}-buildx-
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
```
is there any reason to use `v1` here and `v2` here?
https://github.com/okteto/bin-image/pull/4/files#diff-4f9e38227ed64fefb17f4668a7ac4ab55b6149994d5ac2fd96182d5958479b54R56

pchico83 commented on .github/workflows/deploy.yaml:
```diff
@@ -0,0 +1,81 @@
+          username: "${{ secrets.DOCKER_USERNAME }}"
+          password: "${{ secrets.DOCKER_TOKEN }}"
+
+      - name: Login to GitHub Container Registry
```
Let's just push to Dockerhub. One reason to use GH container registry is unlimited pulls, but we are verified publishers in Dockerhub and our images have unlimited pulls for everyone for free

fdelucchijr replied:
Perfect! I didn't know  about that feature in Dockerhub 👍🏻

pchico83 commented on .github/workflows/deploy.yaml:
```diff
@@ -0,0 +1,81 @@
+on:
+  workflow_dispatch:
+    inputs:
+      bin_tag:
```
This is only executed when there is a new tag, right?

fdelucchijr replied:
workflow_dispatch is a manual trigger, I don't know where can I receive the state of the binaries to dispatch the pipeline 🤔

pchico83 replied:
On other repos we do this:
```
on:
  release:
    types:
      - published
``` 
and then we get the value of the tag like this:
```
    - id: tag
      name: Get the tag
      run: echo ::set-output name=TAG::${GITHUB_REF#refs/tags/}

    - name: "Push"
      run: |
        export TAG=${{ steps.tag.outputs.TAG }}
        make push
```

fdelucchijr replied:
Ohhh! Ok!, I didn't notice your uploading the bin as a release in the repo. That's perfect, I will add the same trigger and removing the workflow_dispatch.

pchico83 commented on .github/workflows/deploy.yaml:
```diff
@@ -0,0 +1,81 @@
+          build-args: |
+            OKTETO_BIN_TAG=${{ github.event.inputs.bin_tag }}
+          tags: |
+            okteto/bin:latest
```
let's remove `latest` here. the tag could be for an older commit than the current master branch

pchico83 reviewed (commented):
@fdelucchijr I love this one! I left some minor comments. Once this is merged, I will reuse your actions for https://github.com/okteto/devenv. Thanks for taking care of this!

fdelucchijr commented on .github/workflows/deploy.yaml:
```diff
@@ -61,9 +55,9 @@ jobs:
           push: true
           build-args: |
-            OKTETO_BIN_TAG=${{ github.event.inputs.bin_tag }}
+            OKTETO_BIN_TAG=${{ github.event.release.tag_name }}
           tags: |
-            okteto/bin:${{ github.event.inputs.bin_tag }}
+            okteto/bin:${{ github.event.release.tag_name }}
           platforms: linux/amd64,linux/arm64
```
This should do the trick based in https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#release

pchico83 reviewed (approved):
Thanks a lot @fdelucchijr!
This was a lot of fun, I am going to use these actions in many places now :-)
